parser_library/src/expressions/data_definition.cpp

```diff
@@ -381,6 +367,62 @@ inline checking::nominal_value_expressions extract_nominal_value_expressions(con
     return values;
 }
 
+constexpr bool is_valid_external_symbol(const context::section& s) noexcept
+{
+    switch (s.kind)
+    {
+        case context::section_kind::DUMMY:
+        case context::section_kind::EXTERNAL_DSECT:
+            return true;
+
+        default:
+            return false;
+    }
```

Comment:
Maybe this if it looks cleaner to you?


```suggestion
    return s.kind == context::section_kind::DUMMY || s.kind == context::section_kind::EXTERNAL_DSECT
```